core/sis-utility/src/main/java/org/apache/sis/xml/Pooled.java

```diff
@@ -260,10 +260,8 @@ final FilterVersion getFilterVersion() {
         switch (xmlnsReplaceCode) {
             case 0: {
                 // Apply namespace replacements only for older versions than the one supported natively by SIS.
-                if (versionGML != null) {
-                    if (versionGML.compareTo(LegacyNamespaces.VERSION_3_2_1) < 0) {
-                        return FilterVersion.GML31;
-                    }
+                if (versionGML != null && versionGML.compareTo(LegacyNamespaces.VERSION_3_2_1) < 0) {
```

Comment:
The reason why this condition was keep in two separated `if` statement is because we currently check for only one version (GML 3.2.1), but we are likely to add new versions in the future. So the code may become something like:

```
if (version GML != null) {
    if (versionGML.compareTo(VERSION_3_2_1) < 0) do something
    if (versionGML.compareTo(VERSION_3_3) >= 0) do something else
    etc.
}
```

Reply:
Going to undo this too.